### PR TITLE
[Verification] Enhanced verifier for end-to-end HF model testing

### PR DESCRIPTION
**Pull request by chhzh123**
## Description ##
This PR follows #87 to enhance the verification support in order to help programmers find bugs in distributed programs in advance. There are several changes:
1. Enhance the consolidation logic for HF models, so that users need not necessarily provide the `_init_weight` function but can still use PyTorch's method to initialize the model.
2. Add `eval_mode` in the argument of the verifier so users can determine whether they want to include `dropout` layers in the verification process.
3. Change the verification logic. First generate the results of the original model and immediately free it to reduce memory usage.
4. Add a test case to verify the TP sharding of the HF BERT model.

## Checklist ##

- [x] PR's title starts with a category (e.g. [Bugfix], [Model], [Tutorial], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

cc @comaniac @zarzen

### CODE REVIEW

File: slapo/build.py [hunk 1]
```diff
@@ -155,7 +155,7 @@ def _consolidate_and_broadcast(sch: Schedule):
                 param.orig_shape if hasattr(param, "orig_shape") else param.shape
             )
             new_param = nn.Parameter(
-                torch.empty(orig_shape, dtype=param.dtype, device=local_rank)
+                torch.zeros(orig_shape, dtype=param.dtype, device=local_rank)
```
[the remark this answers is not in the document]
Comment (chhzh123) [reply slapo/build.py:158]: For this case, the parameter will be uninitialized if using `torch.empty`.
https://github.com/huggingface/transformers/blob/main/src/transformers/models/bert/modeling_bert.py#L693

[the remark this answers is not in the document]
Comment (chhzh123) [reply slapo/build.py:158]: Notice the parameter is also not handled by `_init_weights`, which may lead to possible incorrect results since we first put it on the meta device.
https://github.com/huggingface/transformers/blob/main/src/transformers/models/bert/modeling_bert.py#L748-L762

File: tests/test_verify.py [hunk 2]
```diff
@@ -132,5 +132,73 @@ def forward(self, data):
...
+            subsch["value"].shard("weight", axis=0)
+            subsch["value"].shard("bias", axis=0)
+            fix_attention_mask_shape(subsch)
+            subsch = sch[f"bert.encoder.layer.{i}.attention.output"]
+            subsch["dense"].shard("weight", axis=1)
+            subsch["dense"].sync("fwd_post", sync_op_or_fn="all_reduce")
+            # shard MLP
+            subsch = sch[f"bert.encoder.layer.{i}"]
+            with slapo.Verify(
+                subsch, [torch.randn(bs, seq, config.hidden_size)], enable=False
```
Comment (zarzen) [inline tests/test_verify.py:195]: Why disabling this verification?

Comment (chhzh123) [reply tests/test_verify.py:195]: In general, users only need the outer-most verification context to verify whether the entire schedule is correct or not. If an error is reported, then they can go inside to verify the inner schedules.
